src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/icd/client/DefaultCheckInDelegate.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace app {
+
+CHIP_ERROR DefaultCheckInDelegate::Init(ICDClientStorage * storage)
+{
+    VerifyOrReturnError(storage != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(mpStorage == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    mpStorage = storage;
+    return CHIP_NO_ERROR;
+}
+
+void DefaultCheckInDelegate::OnCheckInComplete(const ICDClientInfo & clientInfo, bool needRefreshKey)
+{
+    mpStorage->StoreEntry(clientInfo);
```

Comment:
Wait, nothing in the OnCheckInComplete API documentation said it was the _callee's_ responsibility to store the entry.  Why is that the callee's responsibility?  And if it is, it _absolutely_ has to be clearly documented.

Reply:
No, this should not be a part of the callback. Discussed with Yunhan offline and he agreed.